packages/shared/routes/setup/Network.svelte

```diff
@@ -0,0 +1,67 @@
+<script lang="typescript">
+    import { localize } from '@core/i18n'
+    import { appRouter } from '@core/router'
+    import { Button, OnboardingLayout, Text } from 'shared/components'
+    import { newProfile, updateActiveProfile } from '@core/profile'
+    import { mobile } from '@lib/app'
+    import { INetwork, NETWORK, NetworkProtocol, NetworkType } from '@core/network'
+    import { TextType } from 'shared/components/Text.svelte'
+
+    const isPrivateNet = (network: INetwork) => network.type === NetworkType.PrivateNet
+
+    const networks: INetwork[] = Object.values(NETWORK).flatMap(Object.values)
+    const filteredNetworks = $newProfile?.isDeveloperProfile
+        ? [...networks.filter((network) => !isPrivateNet(network)), networks.find(isPrivateNet)] // gets only one private net in array
+        : networks.filter((network) => network.type === NetworkType.Mainnet)
+
+    function getNetworkColor(network: INetwork) {
+        if (isPrivateNet(network)) {
+            return 'gray-500'
+        }
+        return network.protocol === NetworkProtocol.Shimmer ? 'shimmer-highlight' : 'black'
+    }
+    function getNetworkIcon(network: INetwork) {
+        return isPrivateNet(network) ? 'settings' : network.protocol
+    }
+    function getNetworkText(network: INetwork) {
+        if (isPrivateNet(network)) {
+            return localize('views.network.privatenet')
+        }
+        return localize(`views.network.${network.protocol}.${network.type}`)
+    }
+    function onClick(network: INetwork): void {
+        updateActiveProfile({ networkProtocol: isPrivateNet(network) ? network.protocol : undefined })
+        updateActiveProfile({ networkType: network.type })
+        $appRouter.next()
```

Comment:
Abstract into an action on the network module, and it will need to be on the newProfile store not the activeProfile store.